File: web-app-service/Core/Gateway/RequestParams.cs
using Core.Reference;

namespace Core.Gateway
{
	public class RequestParams
	{
		public int? Limit { get; set; }

		public int? Offset { get; set; }

		public string SortBy { get; set; }

		public SortDirection? SortDirection { get; set; }
	}
}

File: middleware-service/product-service/core/meta/requestParams.go
package meta

type RequestParams interface {
	Limit() int
	SetLimit(limit int)

	Offset() int
	SetOffset(offset int)

	SortBy() string
	SetSortBy(sortBy string)

	SortDirection() string
	SortDirectionNum() int
	SetSortDirection(direction string)
}